src/components/templates/PartyMap/components/Map/Map.tsx

```diff
@@ -203,9 +203,9 @@ export const Map: React.FC<MapProps> = ({
     () =>
       venue?.rooms
         ?.filter(filterEnabledRooms)
-        .map((room) => (
+        .map((room, index) => (
           <MapRoom
-            key={room.title}
+            key={index}
```

Comment:
See comments in https://github.com/sparkletown/sparkle/pull/1403#issuecomment-841886093 about why we never want to use the array index here.

Reply:
Yes, index is not guaranteed to point to the same room if rooms is "removed" from the list and if sparkle is somehow subscribing to that remove event. How often does that really happen?

If you want "correct" way, then you need to guarantee that `room.title` is always unique (not the case right now) or add another id field (overkill, in my opinion) to the collection. I believe using `index` seems to be an acceptable solution given the context. I don't really think performance is an issue here (have you tried benchmarked it?) unless you are expecting hundreds of rooms to be displayed.

Reply:
An alternative angle to look at this from, is this change required for the main part of your PR fix here?

As this is the existing pattern used across the codebase currently, and since you're only changing one instance of it here (that doesn't seem like it is required to prevent the flickering), then it seems OOS of this PR and would be better raised in a new PR focussed on this (and consistently changed across the entire codebase)

- https://github.com/sparkletown/sparkle/search?q=key%3D%7Broom.title%7D

There are also other variations on this, that combine the `title` with the `index`, which while not perfect, reduces the likelihood of clashes even further (where there would need to be a different room object with the exact same title in the exact same index to cause an issue):

- https://github.com/sparkletown/sparkle/search?q=key%3D%7B%60room-%24%7Broom.title%7D-%24%7Bindex%7D%60%7D
  - ```key={`room-${room.title}-${index}`}```

The `room-` string is superfluous here, so it could just be written as something like:

```
key={`${room.title}-${index}`}
```

This is the general pattern that we use across the codebase in these instances where we don't otherwise have a consistent unique identifier.